widgets/src/c3/Gauge.js

```diff
@@ -0,0 +1,43 @@
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["./Common"], factory);
+    } else {
+        root.Gauge = factory(root.Common);
+    }
+}(this, function (Common) {
+    function Gauge(target) {
+        Common.call(this);
+
+        this._class = "c3_gauge";
+        this._type = "gauge";
+        this._config.data = {
+            columns: []
+        };
+    };
+    Gauge.prototype = Object.create(Common.prototype);
+
+    Gauge.prototype.palette = function (_) {
+        if (!arguments.length) return this._palette;
+        this._config.color.pattern = _;
```

Comment:
I agree, let me look back over the color patterns for this widget... I am
wondering if maybe we should or could standardize the palettes or provide a
helper function?
On Jan 16, 2015 4:53 PM, "Gordon Smith" <notifications@github.com> wrote:

> In widgets/src/c3/Gauge.js
> <https://github.com/hpcc-systems/Visualization/pull/173#discussion-diff-23112229>
> :
>
> > +    }
> > +}(this, function (Common) {
> > +    function Gauge(target) {
> > +        Common.call(this);
> > +
> > +        this._class = "c3_gauge";
> > +        this._type = "gauge";
> > +        this._config.data = {
> > +            columns: []
> > +        };
> > +    };
> > +    Gauge.prototype = Object.create(Common.prototype);
> > +
> > +    Gauge.prototype.palette = function (_) {
> > +        if (!arguments.length) return this._palette;
> > +        this._config.color.pattern = _;
>
> If "_" is one of our Palettes - I don't think this will work. If it isn't
> then we should rename the method name to c3_palette or such like?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcc-systems/Visualization/pull/173/files#r23112229>.
>